src/main/java/edu/harvard/iq/dataverse/search/SolrSearchResult.java

```diff
@@ -560,16 +560,18 @@ public JsonObjectBuilder json(boolean showRelevance, boolean showEntityIds, bool
         }
         
         if (this.entity == null) {
-            
+
         } else {
             if (this.entity.isInstanceofDataset()) {
                 nullSafeJsonBuilder.add("storageIdentifier", this.entity.getStorageIdentifier());
                 Dataset ds = (Dataset) this.entity;
-                DatasetVersion dv;
-                if (this.isDraftState()) {
-                    dv = ds.getLatestVersion();
-                } else {
-                    dv = ds.getReleasedVersion();
+                DatasetVersion dv = null;
+
+                for (DatasetVersion v : ds.getVersions()) {
```

Comment:
This is good, but let's move this logic onto a method on Dataset (in case we need to use it elsewhere some time, and then it makes this fewer lines, ie.
DatasetVersion dv = ds.getVersion(this.datasetVersionId);

Reply:
added the suggested method to Dataset